Skyline.DataMiner.CICD.Tools.DataminerDeploy.Lib/DataMinerService/SLNetDataMinerService.cs

```diff
@@ -0,0 +1,438 @@
+                bool sameVersion = uploadedApp.AppInfo.Version.Equals(version, StringComparison.InvariantCulture);
+
+                if (sameName && sameVersion)
+                {
+                    cleanedCount++;
+                    helper.RemoveUploadedAppPackage(uploadedApp.UploadedPackageID);
+                }
+            }
+
+            return cleanedCount;
+        }
+
+        private string GetAgentVersion()
+        {
+            try
+            {
+                GetAgentBuildInfo buildInfoMessage = new GetAgentBuildInfo();
+                BuildInfoResponse buildInfoResponse = (BuildInfoResponse)slnet.SendSingleResponseMessage(buildInfoMessage);
+
+                if (buildInfoResponse != null && buildInfoResponse.Agents.Length > 0)
+                {
+                    string rawVersion = buildInfoResponse.Agents[0].RawVersion;
+                    return rawVersion;
+                }
+            }
+            catch (DataMinerException e)
+            {
+                logger.LogDebug("(ignoring) Could not retrieve DataMiner Version for requirement checks. Exception: " + e);
+                // best effort, still try the installation.
+            }
+
+            return String.Empty;
+        }
+
+        /// <summary>
+        /// Verifies if the provided version number is higher then the DataMiner Agent version.
+        /// </summary>
+        /// <param name="expectedMinimum">The expected minimum version.</param>
+        /// <param name="version">The version to compare.</param>
+        /// <returns><c>true</c> if the provided version is higher than the specified minimum version number; otherwise, <c>false</c>.</returns>
+        private bool IsVersionHigherOrEqual(string expectedMinimum, string version)
+        {
+            if (expectedMinimum == null || version == null)
+            {
+                return false;
+            }
+
+            if (expectedMinimum == version)
+            {
+                return true;
+            }
+
+            try
+            {
+                int[] versionParts = ParseVersionNumbers(version);
+                int[] minimumParts = ParseVersionNumbers(expectedMinimum);
+
+                for (int i = 0; i < 4; i++)
+                {
+                    int versionPart = versionParts[i];
+                    int minimumVersionPart = minimumParts[i];
+                    if (versionPart > minimumVersionPart)
+                    {
+                        return true;
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                logger.LogDebug("(ignoring) Could not verify DataMiner Minimum Version requirement. Exception: " + e);
+                // best effort, still try the installation.
+                return true;
+            }
+
+            return false;
+        }
+
+        private (string appName, string appVersion, int appBuild) LoadAppPackage(string fullPath)
```

Comment:
appBuild is not used. This can be removed and reduce some parsing.

Reply:
Would prefer to keep it. There is a high chance of this being necessary for pre-releases. It also makes sense to parse it as part of the LoadAppPackage as its part of the appInfo.